lib/internal/parse_args/utils.js

```diff
@@ -0,0 +1,160 @@
+'use strict';
+
+const {
+  ArrayPrototypeFind,
+  ObjectEntries,
+  StringPrototypeCharAt,
+  StringPrototypeIncludes,
+  StringPrototypeSlice,
+  StringPrototypeStartsWith,
+} = primordials;
+
+const {
+  validateObject
+} = require('internal/validators');
+
+// These are internal utilities to make the parsing logic easier to read, and
+// add lots of detail for the curious. They are in a separate file to allow
+// unit testing, although that is not essential (this could be rolled into
+// main file and just tested implicitly via API).
+//
+// These routines are for internal use, not for export to client.
+
+/**
+ * Determines if the argument may be used as an option value.
+ * NB: We are choosing not to accept option-ish arguments.
+ * @example
+ * isOptionValue('V']) // returns true
```

Comment:
Just spotted this random `]` here that I don't think should be here while checking this PR out 😅 
```suggestion
 * isOptionValue('V') // returns true
```

Reply:
Already suggested this change, unsure why it's still there.

@bcoe are the suggested changes for documentation gonna land later on? Or do these have to land [pkgjs/parseargs](https://github.com/pkgjs/parseargs) first?

Reply:
Changes in the code and comments are likely to land in https://github.com/pkgjs/parseargs and get copied down regularly but not instantly. I asked the same question on the other end in the PR with this particular change. 😄

https://github.com/pkgjs/parseargs/pull/91#issuecomment-1094276156

Reply:
I'll do my best to sync this PR with `pkgjs/parseargs` within the next day or so, have been trying to capture any comments here in both places -- sorry if I missed any.